### ISSUE TITLE
Add warning when `--limit` pattern has no matches

### ISSUE THREAD

**Issue by themanifold** [Fizzadar/pyinfra#685]
If you set a `--limit` pattern that has no matches, then there is no warning that there are no matches in the CLI output, and you'll get something like:

```
pyinfra -v --limit "foobar" inventories/something.py
--> Loading config...
--> Loading inventory...
--> Connecting to hosts...

--> Preparing operations...
    Loading: something.py

--> Proposed changes:

--> Beginning operation run...
--> Results:
```
which really gives no clue that you have no matching hosts. If this is done by accident, you might be left groping in the dark (like I was, I set the wrong inventory file in my case in the CLI flag). So just add a little message saying

```
Nothing matched in `--limit` pattern
```
or something similar.

**Comment by Fizzadar** [comment]
Totally agree, treating this as a bug!

**Comment by Fizzadar** [comment]
Just added this in https://github.com/Fizzadar/pyinfra/commit/ac4fee853c9e4c24fa9cd0f8f51576712c722b91 & released in `v1.4.18`.
